src/yaml-file.ts

```diff
@@ -87,13 +97,81 @@ export class YamlFile {
     this.patchOperations.push(...patches);
   }
 
+  /**
+   * A comment to be added to the top of the YAML file.
+   *
+   * Can be multiline. All non-empty line are pefixed with '# '. Empty lines are kept, but not commented.
+   *
+   * For example:
+   * ```ts
+   * pipeline.workflowFile.commentAtTop =
+   * `AUTOGENERATED FILE, DO NOT EDIT!
+   * See ReadMe.md
+   * `;
+   * ```
+   *
+   * Results in YAML:
+   * ```yaml
+   * # AUTOGENERATED FILE, DO NOT EDIT!
+   * # See ReadMe.md
+   *
+   * name: deploy
+   * ...
+   * ```
+   */
+  public set commentAtTop(v: string | undefined) {
+    this._commentBefore = v;
+  }
+
+  public get commentAtTop(): string | undefined {
+    return this._commentBefore;
+  }
```

Comment:
i don't see a use-case for this getter. I think we can just get rid of this and the getter for `commentAtBottom`